File: 2020/2/2.go
package main

import (
	"fmt"
	"bufio"
	"log"
	"os"
	"strconv"
	"strings"
)

func main() {
	file, err := os.Open("input") 
  
	if err != nil { 
		log.Fatalf("failed to open input file") 
	}

	scanner := bufio.NewScanner(file) 
	scanner.Split(bufio.ScanLines) 

	vp := 0
	for scanner.Scan(){
		line := strings.Fields(scanner.Text())
		minmax := strings.Split(line[0], "-")
		min,_ := strconv.Atoi(minmax[0])
		max,_ := strconv.Atoi(minmax[1])
		char := rune(line[1][0])

		cc := 0
		for _, c := range line[2]{
			if(c==char){
				cc++
			}
		}
		if(cc >= min && cc <= max){
			vp++
		}
		
		
		
		fmt.Println(min, max, char, cc, line)
	}

	fmt.Println("vp =", vp)
}

File: 2021/1/2.py
import os
os.chdir('advent_2021/1')
inc = 0
a = []
with open("input.txt", "r") as f:
    for line in f:
        a.append(int(line))
    
p = 0
#print(len(a), p+4, 10 < 4)
while len(a) >= (p + 4):
    n1 = a[p] + a[p+1] + a[p+2]
    n2 = a[p+1] + a[p+2] + a[p+3]
    #print(p, n1, n2, inc)
    if n2 > n1:
        inc += 1
    p += 1
print(inc)
